### DISCUSSION TITLE
single file history only diffs

### DISCUSSION THREAD

**Discussion by gbburkhardt** [gitextensions/gitextensions#10578] [Q&A]
Version 4.0.1.15887
When I bring up the history dialog for a single file, select two commits, and then use the context menu "Compare:Open Selected commits with difftool", I get a comparison of all the files in the two commits, when I really only want that of the file I've selected for the history.  Apparently, there used to be a way to turn off the "full history"  (see [git-extensions-single-file-history-only-diffs](https://stackoverflow.com/questions/32950776/git-extensions-single-file-history-only-diffs) ).  But I can't find that switch anywhere in version 4.  Is it just gone?  I see an option in the code for "fullhistoryinfilehistory" in AppSettings.cs, but can't figure out where it would be displayed (yet!).

Was that feature just lost?

**Comment by mstv** [comment]
Just hit `F3` in order to open the external diff viewer for the selected file.
Or use the file's context menu:

![image](https://user-images.githubusercontent.com/36601201/209873906-8094cd15-72f8-4af3-a56d-fe4ce86f7675.png)

**Comment by gerhardol** [comment]
![image](https://user-images.githubusercontent.com/6248932/209876359-655abe20-7193-426a-a2c9-ae6db395da1d.png)

Open commits with difftool was added in 3.3 or so, maybe you used one of the other options before.
The form mstv show is the recommended way to show diffs though.

**Comment by gbburkhardt** [comment]
It's possible to use 'Remember Second for diff' and 'Remember First for 
diff' to get what I want.  But it's really cumbersome.

If I'm in the file history view for a particular file, I want to be able 
to select two commits from the tree view, and then get a diff of the 
file for which the history is shown.

So, showing the history for a file, select two commits

and then get a diff for that file only for the two commits.

**Reply by mstv** [reply]
> select two commits

`Ctrl+Click`

> and then get a diff for that file only for the two commits.

Hit `F3`
or use the context menu of the *file* (see my screenshot - "the Form mstv has shown")

**Comment by gbburkhardt** [comment]
Sorry, I don't see how this helps.  When displaying the file history, I 
want to be able to select two commits, and see the diff for only the 
file for which the file history was invoked.

What is "mstv show"?
